gno.land/pkg/gnoclient/client_txs.go

```diff
@@ -106,59 +102,73 @@ func (c *Client) Call(cfg BaseTxCfg, msgs ...MsgCall) (*ctypes.ResultBroadcastTx
 	return c.signAndBroadcastTxCommit(tx, cfg.AccountNumber, cfg.SequenceNumber)
 }
 
-// Temporarily load cfg.Package on the blockchain and run main() which can
-// call realm functions and use println() to output to the "console".
-// This returns bres where string(bres.DeliverTx.Data) is the "console" output.
-func (c *Client) Run(cfg RunCfg) (*ctypes.ResultBroadcastTxCommit, error) {
-	// Validate required client fields.
+func (c *Client) Run(cfg BaseTxCfg, msgs ...MsgRun) (*ctypes.ResultBroadcastTxCommit, error) {
+	// Validate required client fields
 	if err := c.validateSigner(); err != nil {
-		return nil, errors.Wrap(err, "validate signer")
+		return nil, err
 	}
 	if err := c.validateRPCClient(); err != nil {
-		return nil, errors.Wrap(err, "validate RPC client")
+		return nil, err
 	}
 
-	memPkg := cfg.Package
-	gasWanted := cfg.GasWanted
-	gasFee := cfg.GasFee
-	sequenceNumber := cfg.SequenceNumber
-	accountNumber := cfg.AccountNumber
-	memo := cfg.Memo
-
-	// Validate config.
-	if memPkg.IsEmpty() {
-		return nil, errors.New("found an empty package " + memPkg.Path)
+	// Validate base transaction config
+	if err := cfg.validateBaseTxConfig(); err != nil {
+		return nil, err
 	}
 
-	// Parse gas wanted & fee.
-	gasFeeCoins, err := std.ParseCoin(gasFee)
-	if err != nil {
-		return nil, errors.Wrap(err, "parsing gas fee coin")
+	// Parse MsgRun slice
+	vmMsgs := make([]vm.MsgRun, 0, len(msgs))
+	for _, msg := range msgs {
+		// Validate MsgCall fields
+		if err := msg.validateMsgRun(); err != nil {
+			return nil, err
+		}
+
+		// Parse send coins
+		send, err := std.ParseCoins(msg.Send)
+		if err != nil {
+			return nil, err
+		}
+
+		caller := c.Signer.Info().GetAddress()
+
+		// Precompile and validate Gno syntax
+		if err = gno.PrecompileAndCheckMempkg(msg.Package); err != nil {
```

Comment:
this is a bit horrible btw, `PrecompileAndCheckMempkg` has the implicit dependency on `gofmt` being installed on the system. This can be fixed, and I made a note to fix it in my upcoming refactoring of `precompile`, but I just wanted to note it in case it causes issues.